dev/integration_tests/semantics/test_driver/android_testing.dart

```diff
@@ -0,0 +1,512 @@
+
+  Map<String, Object> get _flags => _values['flags'];
+
+  void addChild(AndroidSemanticsNode child) => _children.add(child);
+
+  /// The text value of the semantics node.
+  ///
+  /// This is produced by combining the value, label, and hint fields from
+  /// the Flutter [SemanticsNode].
+  String get text => _values['text'];
+
+  /// The className of the semantics node.
+  ///
+  /// Certain kinds of Flutter semantics are mapped to Android classes to
+  /// use their default semantic behavior, such as checkboxes and images.
+  ///
+  /// If a more specific value isn't provided, it defaults to
+  /// "android.view.View".
+  String get className => _values['className'];
+
+  /// The identifier for this semantics node.
+  int get id => _values['id'];
+
+  /// The children of this semantics node.
+  List<AndroidSemanticsNode> get children => _children;
+
+  /// Whether the node is currently in a checked state.
+  ///
+  /// Equivalent to [SemanticsFlag.isChecked].
+  bool get isChecked => _flags['isChecked'];
+
+  /// Whether the node can be in a checked state.
+  ///
+  /// Equivalent to [SemanticsFlag.hasCheckedState]
+  bool get isCheckable => _flags['isCheckable'];
+
+  /// Whether the node is editable.
+  ///
+  /// This is usually only applied to text fields, which map
+  /// to "android.widget.EditText".
+  bool get isEditable => _flags['isEditable'];
+
+  /// Whether the node is enabled.
+  bool get isEnabled => _flags['isEnabled'];
+
+  /// Whether the node is focusable.
+  bool get isFocusable => _flags['isFocusable'];
+
+  /// Whether the node is focused.
+  bool get isFocused => _flags['isFocused'];
+
+  /// Whether the node represents a password field.
+  ///
+  /// Equivalent to [SemanticsFlag.isObscured].
+  bool get isPassword => _flags['isPassword'];
+
+  /// Whether the node is long clickable.
+  ///
+  /// Equivalent to having [SemanticsAction.longPress]/
+  bool get isLongClickable => _flags['isLongClickable'];
+
+  /// Gets a [Rect] which defines the position and size of the semantics node.
+  Rect getRect() {
+    final Map<String, int> rect = _values['rect'];
+    return new Rect.fromLTRB(
+      rect['left'].toDouble(),
+      rect['top'].toDouble(),
+      rect['right'].toDouble(),
+      rect['bottom'].toDouble(),
+    );
+  }
+
+  /// Gets a [Size] which defines the size of the semantics node.
+  Size getSize() {
+    final Rect rect = getRect();
+    return new Size(rect.bottom - rect.top, rect.right - rect.left);
+  }
+
+  /// Gets a list of [AndroidSemanticsActions] which are defined for the node.
+  List<AndroidSemanticsAction> getActions() {
+    final List<AndroidSemanticsAction> result = <AndroidSemanticsAction>[];
+    for (int id in _values['actions']) {
+      result.add(new AndroidSemanticsAction._(id));
+    }
+    return result;
+  }
+
+  @override
+  String toString() {
+    return _values.toString();
+  }
+}
+
+/// Matches an [AndroidSemanticsNode].
```

Comment:
This should probably point out how it is different from hasSemantics and mention that this compares with what the native Android side things the semantics are.